tests/test_dataclasses.py

```diff
@@ -2547,3 +2547,28 @@ class Model(BaseModel, Base):
         pass
 
     assert Model().a == 5
+
+
+def test_alias_with_dashes():
+    """Test for fix issue #7226."""
+    # WITH
+    data = {'some-var': 'some_value'}
+
+    @pydantic.dataclasses.dataclass
+    class Example:
+        some_var: str = Field(..., alias='some-var')
+
+    @pydantic.dataclasses.dataclass(frozen=True)
```

Comment:
Why do we need to test this for frozen dataclasses?

Reply:
We don't need but in the reported issue it was.  :D I will remove it.